File: models/recipe.go
package models

// Recipe defines a cooking recipe
type Recipe struct {
	BaseModel
	Title string `gorm:"unique;not null" json:"title"`
	Description string `json:"description"`
	UsedProducts string `validate:"required" json:"usedProducts"`
	Picture string `json:"picture"`
	CookingTime int `validate:"required" json:"cookingTime"`
	UserID ModelID `json:"userid"`
	User User `json:"user"`
	CategoryID ModelID `json:"categoryid"`
	Category Category `json:"category"`
	Comments []Comment `json:"comments"`
}

File: client-app/src/app/models/recipe.ts
import { ICategory } from "./category";

export interface IRecipe {
    id: string;
    title: string;
    description: string;
    usedProducts: string;
    cookingTime: Number;
    category: ICategory;
    categoryId: string;
    user?: string;
}

export interface IRecipeFormValues extends Partial<IRecipe> {

}

export class RecipeFormValues implements IRecipeFormValues {
    id?: string = undefined;
    title: string = '';
    category?: ICategory = undefined;
    description: string = '';
    usedProducts: string = '';
    cookingTime: Number = 0;
    user?: string = '';

    constructor(init?: IRecipeFormValues) {
        Object.assign(this, init);
    }
}